modules/pipeline/providers/reconciler/legacy/reconciler/listen.go

```diff
@@ -133,44 +78,46 @@ func (r *Reconciler) doWatch(ctx context.Context, key string, t storetypes.Chang
 	}()
 
 	// reconciler
-	reconcileErr := r.reconcile(pCtx, pipelineID)
+	reconcileErr := r.internalReconcile(pCtx, pipelineID)
 
 	defer func() {
 		r.teardownCurrentReconcile(pCtx, pipelineID)
 		if err := r.updateStatusAfterReconcile(pCtx, pipelineID); err != nil {
 			rlog.PErrorf(pipelineID, "failed to update status after reconcile, err: %v", err)
 		}
 
-		// if reconcile failed, put and wait next reconcile
+		// if reconcile failed, put and wait next Reconcile
 		if reconcileErr != nil {
-			rlog.PErrorf(pipelineID, "failed to reconcile pipeline, err: %v", err)
+			rlog.PErrorf(pipelineID, "failed to reconcile pipeline, err: %v", reconcileErr)
 			r.reconcileAgain(pipelineID)
 		}
 	}()
 }
 
-// reconcileAgain add to reconciler again, wait next reconcile
+// reconcileAgain add to reconciler again, wait next Reconcile
 func (r *Reconciler) reconcileAgain(pipelineID uint64) {
 	time.Sleep(time.Second * 5)
-	// delete the recorded pipeline that is being executed to prevent failure to reconcile
+	// delete the recorded pipeline that is being executed to prevent failure to Reconcile
 	r.processingPipelines.Delete(pipelineID)
-	r.Add(pipelineID)
+	for r.internalReconcile(context.Background(), pipelineID) != nil {
+		time.Sleep(time.Second * 5)
```

Comment:
why sleep 5 again?

Reply:
because it's continue reconcile again,  and sleep 5s first and do internalReconcile inside for-loop. if failed at for-loop, sleep 5s and try again.

Reply:
I changed code to invoke `r.Reconcile`, the replacement of `r.Add`.

Reply:
adjusted to another format.